### PR TITLE
updated the Monolog provider to use the newest features of Monolog and the Symfony bridge

### PR DESCRIPTION
**Pull request by fabpot**
@Seldaek If you can have a quick look, that would be much appreciated :)

### CODE REVIEW

File: src/Silex/Provider/MonologServiceProvider.php [hunk 1]
```diff
@@ -77,14 +98,19 @@ public function register(Container $app)
...
         $app['monolog.permission'] = null;
         $app['monolog.exception.logger_filter'] = null;
+        $app['monolog.logfile'] = null;
     }
 
     public function boot(Application $app)
     {
+        if (!$app['debug']) {
+            ErrorHandler::register($app['monolog']);
+        }
```
Comment (Seldaek) [inline src/Silex/Provider/MonologServiceProvider.php]: Do you really mean to only enable this in non-debug mode? I am not familiar enough with silex to know how it behaves internally in debug mode so I'm not sure.

**Comment by Seldaek** [comment]
I think it looks good, except for the fact that monolog.handler (handler) becomes monolog.handlers (array), which is a BC break right? Might be nice to define `monolog.handlers` as `(array) $app['monolog.handler']` in case it is set, but I don't know how much you care about BC.

File: src/Silex/Provider/MonologServiceProvider.php (hunk 1 above)
Comment (henrikbjorn) [inline src/Silex/Provider/MonologServiceProvider.php]: we have had some errors where the ErrorHandler muted exceptions etc. should be optional imo.

Comment (Seldaek) [inline src/Silex/Provider/MonologServiceProvider.php]: Could that maybe be fixed?

Comment (fabpot) [inline src/Silex/Provider/MonologServiceProvider.php]: Not sure why I did that (I extracted the code from an old project). Let's remove the condition and see if that works for everyone.

**Pull request merged**